nservicebus/best-practices.md

```diff
@@ -0,0 +1,130 @@
+---
+title: Best practices
+summary: An assortment of best practices presented as DO, DO NOT, and CONSIDER.
+reviewed: 2020-10-01
+---
+
+This article will present recommendations to keep in mind when designing a system using NServiceBus.
+
+## Architecture
+
+
+:x: **DO NOT abstract away NServiceBus**
+
+NServiceBus is itself an abstraction over different message queue technologies, allowing message handlers to concern themselves with business code and not worry about the speific implementation of the underlying queue.
+
+As a dependency, conventional wisdom holds that the NServiceBus API should itself be abstracted behind another abstraction, perhaps as a hedge against a possible future where the system will have to completely switch to a different service bus library.
+
+However, NServiceBus is a fairly opinionated framework that attempts to guide developers building distributed systems into a [pit of success](https://blog.codinghorror.com/falling-into-the-pit-of-success/). It also becomes the communication foundation for the entire system.
+
+Because of these qualities, abstractions over NServiceBus tend to basically mimic the NServiceBus API which add no value. When it comes time to switch to a different service bus framework (which is a dreadfully uncommon activity to start with as it does not deliver any business value) it ends up being much more difficult than simply swapping one implementation for a new one.
+
+Meanwhile, a custom abstraction makes this NServiceBus documentation fairly useless as developers must code to a custom, often undocumented API instead of directly against the NServiceBus API.
+
+:x: **DO NOT use NServiceBus to query data**
```

Comment:
You might want to review this one

Reply:
Oh, crap, you already wrote something about this at the bottom. Delete it if you will.